test/ref_ops_test.cpp

```diff
@@ -1849,6 +1849,40 @@ TEST_CASE(cosh_dyn_test)
     EXPECT(migraphx::verify_range(results_vector, gold));
 }
 
+TEST_CASE(convert_nan_upcast_test)
```

Comment:
Add a test case for half as well and even a double->float->half to ensure things stay preserved.

Reply:
I don't see the need to add a Double convert test (a) it is very unlikely to have triple converts inside a model. (b) Double converts are just two convert operations in tandem. I've individual convert tests, if they work, then double converts would work for sure. 

For the GPU target, we run simplify_algebra/reshapes. Which would convert such double converts to single convert. It doesn't happen for "ref" though.  

I added extra two tests.